### PR TITLE
Merge smart_proxy_acd_core into smart_proxy_acd

### PR DESCRIPTION
**Pull request by adamruzicka**
Part of stage 3 of [phasing out smart proxy dynflow core proposal](https://community.theforeman.org/t/phasing-out-smart-proxy-dynflow-core/22841)

Once this gets in then:
- the next released version of `smart_proxy_acd` will need to deprecate `smart_proxy_acd_core` in packaging
- we will be able to drop `smart_proxy_acd_core` from packaging

Sadly rubocop was unhappy about compact style module definitions, which forced me to add an additional level of indentation, leading to a much larger diff than what is functionally necessary.

### CODE REVIEW

**Comment by adamruzicka** [comment]
Disclaimer: I currently don't have a working setup with ACD, there may be dragons

**Comment by ehelms** [comment]
@sbernhard Can you please review this? We need a release with this fix in it to fix nightly upgrades.

File: lib/smart_proxy_acd/acd_runner.rb [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+          certificate_file = @dynflow_settings['foreman_ssl_cert'] || @dynflow_settings['ssl_certificate']
+          if certificate_file
+            certificate = File.read(certificate_file)
+            @ssl_options[:ssl_client_cert] = OpenSSL::X509::Certificate.new(certificate)
+          end
+          ca_file = @dynflow_settings['foreman_ssl_ca'] || @dynflow_settings['ssl_ca_file']
+          @ssl_options[:ssl_ca_file] = ca_file if ca_file
+          @ssl_options
+        end
+      end
```
Comment (ekohl) [inline lib/smart_proxy_acd/acd_runner.rb:49]: Is this still needed? The Smart Proxy has options for this. Like `Proxy::SETTINGS.foreman_url`. Even better may be to use [ForemanRequest](https://github.com/theforeman/smart-proxy/blob/develop/lib/proxy/request.rb)

File: smart_proxy_acd.gemspec [hunk 2]
```diff
@@ -14,4 +14,5 @@ Gem::Specification.new do |s|
   s.license = 'GPLv3'
 
   s.add_development_dependency('rubocop', '0.50.0')
+  s.add_runtime_dependency('smart_proxy_dynflow', '>= 0.5.0')
```
Comment (ekohl) [inline smart_proxy_acd.gemspec:17]: Should it also depend on REX as the previous core did or is that no longer needed?

File: lib/smart_proxy_acd/acd.rb [hunk 3]
```diff
@@ -7,6 +7,14 @@ class Plugin < ::Proxy::Plugin
       plugin 'acd', Proxy::Acd::VERSION
       http_rackup_path File.expand_path('acd_http_config.ru', File.expand_path('../', __FILE__))
       https_rackup_path File.expand_path('acd_http_config.ru', File.expand_path('../', __FILE__))
+
+      after_activation do
```
Comment (ekohl) [inline lib/smart_proxy_acd/acd.rb]: Same as https://github.com/theforeman/smart_proxy_salt/pull/64#discussion_r666227288

File: smart_proxy_acd.gemspec (hunk 2 above)
Comment (adamruzicka) [reply smart_proxy_acd.gemspec:17]: Not sure why it was there in the first place, from what I've seen it didn't seem to be using anything from rex core

File: lib/smart_proxy_acd/acd.rb [hunk 4]
```diff
@@ -5,8 +5,18 @@ class NotFound < RuntimeError; end
     # Implement a SmartProxy Plugin
     class Plugin < ::Proxy::Plugin
       plugin 'acd', Proxy::Acd::VERSION
-      http_rackup_path File.expand_path('acd_http_config.ru', File.expand_path('../', __FILE__))
-      https_rackup_path File.expand_path('acd_http_config.ru', File.expand_path('../', __FILE__))
+      http_rackup_path File.expand_path('acd_http_config.ru', __dir__)
+      https_rackup_path File.expand_path('acd_http_config.ru', __dir__)
```
Comment (ekohl) [inline lib/smart_proxy_acd/acd.rb]: It gets unrelated, but with recent Smart Proxy versions (2.3+) you can use
```suggestion
      rackup_path File.expand_path('acd_http_config.ru', __dir__)
```

**Review by ekohl — APPROVED** [review]

File: lib/smart_proxy_acd/acd_runner.rb (hunk 1 above)
Comment (sbernhard) [reply lib/smart_proxy_acd/acd_runner.rb:49]: thanks for this idea. we should handle this with another PR.

**Comment by sbernhard** [comment]
Thank you @adamruzicka and @ekohl

**Pull request merged**